pkg/controller/useraccount/useraccount_controller_test.go

```diff
@@ -1133,6 +1220,164 @@ func TestDisabledUserAccount(t *testing.T) {
 	})
 }
 
+func TestLookupAndDeleteCheUser(t *testing.T) {
+	// given
+	username := "sugar"
+	userID := uuid.NewV4().String()
+
+	t.Run("che user deletion is not enabled", func(t *testing.T) {
+		// given
+		userAcc := newUserAccount(username, userID, false)
+		r, _, _ := prepareReconcile(t, username, userAcc, cheRoute(true), keycloackRoute(true))
+
+		// when
+		err := r.lookupAndDeleteCheUser(userAcc)
+
+		// then
+		require.NoError(t, err)
+	})
+
+	t.Run("che user deletion is enabled", func(t *testing.T) {
+		memberOperatorSecret := newSecretWithCheAdminCreds()
+		restore := test.SetEnvVarsAndRestore(t,
+			test.Env("WATCH_NAMESPACE", "toolchain-member"),
+			test.Env("MEMBER_OPERATOR_SECRET_NAME", "test-secret"),
+			// Che has two different routes for Che & Keycloak but CRW uses a single route for both.
+			// For tests we'll assume they're two different routes.
+			test.Env(configuration.MemberEnvPrefix+"_CHE_KEYCLOAK_ROUTE_NAME", "keycloak"),
+		)
+		defer restore()
+
+		t.Run("get token error", func(t *testing.T) {
+			// given
+			defer gock.OffAll()
+			gockTokenFail()
+			gockFindUserSuccess(username)
+			gockDeleteUser(204)
```

Comment:
if the token retrieval fails, then you don't need these mocks
```suggestion
```

Reply:
True. Fixed.